### ISSUE TITLE
Report query-level invalid accumulation errors in `ReduceInaccumulable`

### ISSUE THREAD

**Issue by vmarcos** [MaterializeInc/materialize#18212]
Currently, basic aggregates check for invalid accumulations and produce output in the logs as well as omit results:
https://github.com/MaterializeInc/materialize/blob/bae5ea5cfe28408bad1fe64e99a17b16803f099d/src/compute/src/render/reduce.rs#L572-L585
This issue is a request to improve this behavior to not only log errors to Sentry, but also produce clean query-level errors in this situation. 

The following reproduction of the error builds on `test/testdrive/negative-multiplicities.td` and requires us to run Materialize with `--unsafe-mode`. Note that the latter is required to make it easy to create the data with too many retractions; it is also possible to trigger the behaviors below if, e.g., too many retractions get into a source outside of unsafe mode.

First, set some test data up:
```sql
CREATE TABLE base (data bigint, diff bigint);
CREATE MATERIALIZED VIEW data AS SELECT data FROM base, repeat_row(diff);
INSERT INTO base VALUES (1, 1);
INSERT INTO base VALUES (1, -1), (1, -1);
INSERT INTO base VALUES (2, 1);
```

Then, you should be able to observe the following:

```sql
materialize=> SELECT list_agg(data) FROM data;
 list_agg 
----------
 
(1 row)
```
The above command produces the following output in the logs:
```
cluster-u1-replica-1: 2023-03-17T12:24:48.566944Z  WARN mz_compute::render::reduce: [customer-data] Negative accumulation in ReduceInaccumulable: Row{[List([List([Int64(1)])])]} with count -1 in dataflow temp-view-t34
cluster-u1-replica-1: 2023-03-17T12:24:48.567008Z ERROR mz_compute::render::reduce: Negative accumulation in ReduceInaccumulable
```
We would like, however, to see it produce a query-level error in addition to the log output.

**Comment by vmarcos** [comment]
The solution approach to this issue could employ a `reduce_pair`, similarly to what was done in PR #17990.

**Comment by tokenrove** [comment]
Closed by #18502.
